### PR TITLE
Adding Space Components and add ICT equipments

### PR DESCRIPTION
**Pull request by alinamstanciu**
- Added Space components as done by Rick in this PR https://github.com/WillowInc/opendigitaltwins-building/commit/c3eba4045e41527a2421a9fe68b671aab0564234
- Add more ICT equipment as done by Rick in this PR https://github.com/WillowInc/opendigitaltwins-building/commit/d672c28b9ddfaaf8a3100015fd88741a4cb1ef9d

Rick, next time when you add something generic, I think it's worth pinging us if it could sit in this main repo

### CODE REVIEW

**Comment by akshayj-MSFT** [comment]
The folders under Asset\Equipment have the word "Equipment" tacked onto each folder - this seems overkill. 
I prefer the Willow way which has
Equipment
- Conveyance
- Electrical
- Fire Protection
- HVAC
- ICT
- Lighting
- etc

Would it be possible to make a similar change as part of this PR?

File: Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment/BACnetController.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "@type": "Interface",
+    "displayName": {
+      "en": "BACnet Controller"
+    },
+    "extends" : [
+      "dtmi:digitaltwins:rec_3_3:asset:ControllerEquipment;1"
+    ],
+    "contents": [
+      {
+        "@type": "Property",
```
Comment (rszcodronski) [inline Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment/BACnetController.json:12]: I would remove this `address` property as it is a Willow specific implementation detail.

File: Ontology/Asset/Equipment/ICTEquipment/GatewayEquipment.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{
+  "@id": "dtmi:digitaltwins:rec_3_3:asset:GatewayEquipment;1",
```
Comment (rszcodronski) [inline Ontology/Asset/Equipment/ICTEquipment/GatewayEquipment.json:2]: Why do we need to append Equipment?

File: Ontology/Asset/Equipment/ICTEquipment/GatewayEquipment.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  "@type": "Interface",
+  "displayName": {
+    "en": "Gateway"
+  },
+  "extends": "dtmi:digitaltwins:rec_3_3:asset:ICTEquipment;1",
+  "contents": [
```
Comment (rszcodronski) [inline Ontology/Asset/Equipment/ICTEquipment/GatewayEquipment.json:8]: Similar to Controller, I would remove all of these Properties.

File: Ontology/Asset/Equipment/ICTEquipment/SensorEquipment/PeopleCountSensor/PointGrabCogniPointSensor.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{
+  "@id": "dtmi:digitaltwins:rec_3_3:asset:PointGrabCogniPointSensor;1",
```
Comment (rszcodronski) [inline Ontology/Asset/Equipment/ICTEquipment/SensorEquipment/PeopleCountSensor/PointGrabCogniPointSensor.json:2]: I wouldn't include this as it is a vendor-specific model.

**Review by rszcodronski — COMMENTED** [review]
Also consider adding the Space Components to the following:
* TenantUnit

File: Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
+{
+  "@id": "dtmi:digitaltwins:rec_3_3:asset:ControllerEquipment;1",
```
[the remark this answers is not in the document]
Comment (akshayj-MSFT) [reply Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment.json:2]: @alinamstanciu see my uber comment above about "Equipment"
@rszcodronski  fyi

[the remark this answers is not in the document]
Comment (alinamstanciu) [reply Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment.json:2]: It's not completely true, all equipment under Equipment folder has suffix "Equipment", I think Karl decided to go this way, I agree that is overkill, so I am going to remove it first from the files I am changing, in a different PR from all equipments

File: Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  "@type": "Interface",
+  "displayName": {
+    "en": "Controller"
+  },
+  "extends": "dtmi:digitaltwins:rec_3_3:asset:ICTEquipment;1",
+  "contents": [
```
[the remark this answers is not in the document]
Comment (alinamstanciu) [reply Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment.json:8]: Done

File: Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment/BACnetController.json (hunk 1 above)
Comment (alinamstanciu) [reply Ontology/Asset/Equipment/ICTEquipment/ControllerEquipment/BACnetController.json:12]: Done

File: Ontology/Asset/Equipment/ICTEquipment/GatewayEquipment.json (hunk 2 above)
Comment (alinamstanciu) [reply Ontology/Asset/Equipment/ICTEquipment/GatewayEquipment.json:2]: Removed

File: Ontology/Asset/Equipment/ICTEquipment/GatewayEquipment.json (hunk 3 above)
Comment (alinamstanciu) [reply Ontology/Asset/Equipment/ICTEquipment/GatewayEquipment.json:8]: Done

**Comment by alinamstanciu** [comment]
Hi @rszcodronski and @akshayj-MSFT , could you please revisit? Addressed all your feedback
Thanks

**Comment by akshayj-MSFT** [comment]
@alinamstanciu 
Top level Equipment interfaces should go inside their respective folder.
e.g. ConveyanceEquipment,json should move to Conveyance folder
same for ElectricalEquipment, FireProtectionEquipment, etc, etc, etc

**Review by akshayj-MSFT — APPROVED** [review]
Looks good Alina! Exciting additions!

**Review by rszcodronski — COMMENTED** [review]
I believe the files get placed into the appropriate folders from the REC owl-dtdl converter so if they are changed this would break.

Everything else looks good.

**Comment by hammar** [comment]
Status update: I've set aside Friday for work on this (and on the externalIds feature) and will return the review then. Pardon the delay.

**Comment by alinamstanciu** [comment]
@rszcodronski , any more feedback I need to address, if not, could you please signoff? Thanks

**Review by rszcodronski — APPROVED** [review]
Approved with caveat that we need to check that the file/folder structure won't break when run through the owl-to-dtdl converter

**Comment by alinamstanciu** [comment]
I haven't made any changes to the files/folders yet @rszcodronski

**Review by hammar — COMMENTED** [review]
I'm not in agreement with the folder renaming, which has taken nouns and turned them into adjective or verb-like names that are in some cases ambiguous (e.g., Fire Protection, Conveyance, etc.). Note that these folders are automatically generated to correspond with the super-interface in question by OWL2DTDL, so they will be returned to the previous name automatically.

The BACnetController and ModbusController interfaces make sense, provided that these are physical boxes, rather than logical/software functionalities (I don't know the domain well enough to tell).

The ICTHardware interface is inconsistent with the file name (ITCHardware.json), but I'm OK with the interface name. When generating the interface from OWL2DTDL this will be fixed.

All of the ICT/SensorEquipment subinterfaces seem to make sense, again, under the assumption that we are here talking about physical boxes rather than capabilities; the dualism between capabilities and equipments does seem a little inelegant, but I can see it makes sense, and you guys have had long discussions on this that I have not been able to keep up-to-date on.

All in all: I'm ready to backport most of this into OWL-REC, and regenerate using OWL2DTDL (and at that point ideally also get the externalIds property on top interfaces into place), but this will revert the folder names. OK with all?

UPDATE: I didn't see all the space components -- will review them also tonight CET and update this comment thereafter.

**Pull request merged**